File: C++/2022. Convert 1D Array Into 2D Array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (original.size() != m * n) return {}; 
        vector<vector<int>> ans(m, vector<int>(n)); 
        for (int i = 0, k = 0; i < m; i++) 
            for (int j = 0; j < n; ++j, ++k) 
                ans[i][j] = original[k]; 
        return ans; 
    }
};

File: JAVA/2022. Convert 1D Array Into 2D Array.java
class Solution {
    public int[][] construct2DArray(int[] a, int m, int n) {
        if(m*n!=a.length){
            return new int[0][0];
        }
        int[][] res=new int[m][n];
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                res[i][j]=a[n*i +j];
            }
        }
        return res;
    }
}
